inputFiles/main.ats

```diff
@@ -0,0 +1,21 @@
+"""
+Primary ATS script
+"""
+import os
+import ats
+from pathlib import Path
+
+# Set default ats parameters
+glue(collection='')
+glue(checks='')
+glue(solvers='')
+glue(outputs='')
+glue(constitutive_models='')
+
+# Recursively search the child directories and add them to ATS
+root_path = Path(ats.tests.AtsTest.getOptions().get("atsRootDir"))
+Nr = len(root_path.parts)
+for path in root_path.rglob('*.ats'):
+    if (path.name != 'main.ats'):
+        glue(collection=path.parts[Nr + 1])
+        source(path.absolute())
```

Comment:
```suggestion
def main():
    # Set default ats parameters
    glue(collection='')
    glue(checks='')
    glue(solvers='')
    glue(outputs='')
    glue(constitutive_models='')

    # Recursively search the child directories and add them to ATS
    root_path = Path(ats.tests.AtsTest.getOptions().get("atsRootDir"))
    Nr = len(root_path.parts)
    for path in root_path.rglob('*.ats'):
        if (path.name != 'main.ats'):
            glue(collection=path.parts[Nr + 1])
            source(path.absolute())


if __name__ == "__main__":
    main()          
```
Does it make sense? Is it working or is there black magic?

Reply:
I'm not entirely sure how ATS calls these files, so I'm not sure how it would react to a __main__ guard